zstacklib/zstacklib/test/test_convert_volume_encryption.py

```
@@ -0,0 +1,167 @@
+            linux.convert_qcow2_volume_encryption(self.src, self.dst, False)
+
+        self.assertEqual([self.dst], self.convert_targets)
+        self.assertFalse(os.path.exists(self.dst))
+        with open(self.src, "rb") as fd:
+            self.assertEqual("source", fd.read())
+
+    def test_keeps_existing_file_when_validation_fails(self):
+        with open(self.dst, "wb") as fd:
+            fd.write("existing")
+
+        with self.assertRaises(Exception):
```

Ruff (0.15.21)

[warning] Do not assert blind exception: `Exception`

(B017)